lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
```

Comment:
In Puppet there is: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows.rb#L2-L8 for the sake of tests.  You could do something like this (i.e. define `set_symlink_folder` and then redefine if on Windows).

Or you could do something similar to what was done in sqlserver
https://github.com/puppetlabs/puppetlabs-sqlserver/blob/36a45a89df43f8802a9882d87443bc43ec86d74b/lib/puppet_x/sqlserver/sql_connection.rb